crates/sui-indexer-alt-jsonrpc/src/api/transactions/response.rs

```diff
@@ -134,9 +128,9 @@ fn effects(tx: &StoredTransaction) -> Result<SuiTransactionBlockEffects, RpcErro
 async fn events(
     ctx: &Context,
     digest: TransactionDigest,
-    tx: &StoredTransaction,
+    tx: &TransactionContents,
 ) -> Result<SuiTransactionBlockEvents, RpcError<Error>> {
-    let events: Vec<Event> = bcs::from_bytes(&tx.events).context("Failed to deserialize Events")?;
+    let events: Vec<Event> = tx.events()?;
```

Comment:
nit: probably don't need the type annotation anymore

```suggestion
    let events = tx.events()?;
```